agent/hcp/client/metrics_client.go

```diff
@@ -150,8 +154,18 @@ func (o *otlpClient) ExportMetrics(ctx context.Context, protoMetrics *metricpb.R
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("failed to export metrics: code %d: %s", resp.StatusCode, string(body))
```

Comment:
Note: `body` here is actually a reference to the request body instead of the response body so fixing that up with `respData.String()`

Reply:
whoops 😓 thanks, good catch!